libs/vkd3d/va_map.c

```diff
@@ -288,16 +292,44 @@ VkAccelerationStructureKHR vkd3d_va_map_place_acceleration_structure(struct vkd3
         }
     }
 
-    key.view_type = VKD3D_VIEW_TYPE_ACCELERATION_STRUCTURE;
+    key.view_type = VKD3D_VIEW_TYPE_ACCELERATION_STRUCTURE_OR_OPACITY_MICROMAP;
     key.u.buffer.buffer = resource->vk_buffer;
     key.u.buffer.offset = va - resource->va;
     key.u.buffer.size = resource->size - key.u.buffer.offset;
     key.u.buffer.format = NULL;
 
-    view = vkd3d_view_map_create_view(view_map, device, &key);
+    view = vkd3d_view_map_create_view2(view_map, device, &key, rtas_is_omm);
     if (!view)
-        return VK_NULL_HANDLE;
-    return view->vk_acceleration_structure;
+        return;
+    
+    if (view->info.buffer.rtas_is_micromap)
```

Comment:
Is the implication here that is a VA has been used for RTAS, it can never be used for OMM and vice versa?

Reply:
Oh hmm, that's tricky. I don't think D3D12 forbids it, so I guess we'll need some mechanism that destroys existing AS/OMM and creates the other one if we receive a call in a context where we are sure that VA is supposed to be AS or OMM, like `BuildRaytracingAccelerationStructure`.

I'll think about how to address this… meanwhile it would be nice to have tests that verify expected behavior :slightly_smiling_face:

Reply:
Added some FIXMEs for now.